api/src/test/java/org/openmrs/module/fhir2/providers/r4/MedicationRequestFhirResourceProviderTest.java

```diff
@@ -73,4 +89,97 @@ public void getMedicationRequestByUuid_shouldThrowResourceNotFoundException() {
 		MedicationRequest medicationRequest = resourceProvider.getMedicationRequestByUuid(id);
 		assertThat(medicationRequest, nullValue());
 	}
+
+	
+	@Test
+	public void searchMedicationRequest_shouldReturnMatchingMedicationRequestUsingCode() {
+	
+		when(fhirMedicationRequestService.searchForMedicationRequests(any(), any(), any(), any(), any()))
+				.thenReturn(new BaseFhirIBundleResourceProviderTest<>(Collections.singletonList(medicationRequest), 10, 1));
+		
+		TokenAndListParam code = new TokenAndListParam();
+		TokenParam codingToken = new TokenParam();
+		codingToken.setValue("1000");
+		code.addAnd(codingToken);
+		
+		IBundleProvider results = resourceProvider.searchForMedicationRequests(null, null, null, code, null, null);
+		assertThat(results, notNullValue());
+		assertThat(results.getResources(1, 5), hasSize(equalTo(1)));
+		assertThat(results.getResources(1, 5).get(0), notNullValue());
```

Comment:
@reagan-meant  Can we have a get() method which just gets the resources once instead of querying the database repeatedly? Please use this comment throughout the PR.

Reply:
Let me do so @varung-31